### PR TITLE
Save/Restore SCK pin state when SPI is stopped and started before transaction

### PR DESCRIPTION
**Pull request by bugobliterator**
This fixes an issue where SPI bus lines go into undefined state right before the transaction starts, as shown below. This led to occasional spurious transactions with ICM45686 at startup. The fix was to simply hold the SCK line HIGH for CPOL==1 and LOW for CPOL==0.

Before:

After:

* Tested on CubeOrangePlus with and without SPI mode changes during runtime, from MODE0 to MODE3
* Also fixed IMU3 (ICM45686) spi mode on CubeOrangePlus, it should be MODE0 or MODE3. Its not set to MODE0 instead of MODE1.

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/SPIDevice.cpp [hunk 1]
```diff
@@ -380,11 +381,24 @@ bool SPIDevice::acquire_bus(bool set, bool skip_cs)
         bus.spicfg.cr1 = (uint16_t)(freq_flag | device_desc.mode);
         bus.spicfg.cr2 = 0;
 #endif
+        // read current pin mode of sck
+        const iomode_t sck_mode = palReadLineMode(spi_devices[device_desc.bus].sck_line);
+        if (device_desc.mode == SPIDEV_MODE0 || device_desc.mode == SPIDEV_MODE1) {
+            // Clock polarity is 0, so we need to set the clock line low before spi reset
+            palClearLine(spi_devices[device_desc.bus].sck_line);
+            palSetLineMode(spi_devices[device_desc.bus].sck_line, PAL_MODE_OUTPUT_PUSHPULL);
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/SPIDevice.cpp:389]: minor tweak, the palSetLineMode can be outside the if()

**Review by tridge — CHANGES_REQUESTED** [review]
I think we need this in SPIBus::dma_deallocate() too
It would be better to have a common function that does spiStop() and the palSetLineMode() that is called from both places, replacing current calls to spiStop()
maybe call it stop_peripheral() ?

**Review by andyp1per — APPROVED** [review]
Really interesting - this is quite similar to some of the stuff I have had to do on F103 to avoid exactly these kinds of glitches when switching from output to input and back again

File: libraries/AP_HAL_ChibiOS/SPIDevice.cpp (hunk 1 above)
Comment (bugobliterator) [reply libraries/AP_HAL_ChibiOS/SPIDevice.cpp:389]: This shouldn't be needed in spiStop under dma_deallocate, as that is after the transaction has finished and very unlikely to cause issue.

Comment (tridge) [reply libraries/AP_HAL_ChibiOS/SPIDevice.cpp]: it is needed, as it impacts the next transaction

Comment (bugobliterator) [reply libraries/AP_HAL_ChibiOS/SPIDevice.cpp]: ok adding

**Review by tridge — APPROVED** [review]

**Pull request merged**